### PR TITLE
Add coinbase receiver to status, allow it to be changed by an advanced command

### LINKED ISSUE #8125: Feature: add coinbase receiver info to mina client status
At the moment, there is no way to display the current coinbase receiver through the CLI.
It could be useful to see it by typing `mina client status`. It would allow to double check if it's set properly in the daemon and is active. e.g. under the "block producers running" field:

```bash
Block producers running:                       1 (<block_producer_pubkey>)
Coinbase receiver:                             <coinbase_receiver_pubkey>
Best tip consensus time:                       epoch=0, slot=2633
```

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR adds the 'coinbase receiver' optional field to status, and adds a `advanced set-coinbase-receiver` command.

Currently, there is no way to find out this information without waiting for a block to be produced and manually inspecting it.

Sample output:
```
$ _build/default/src/app/cli/src/mina.exe client status
Mina daemon status
-----------------------------------

Max observed block height:              1
Max observed unvalidated block height:  0
Local uptime:                           9s
Chain id:                               971ab7c2a9370f90a3e4f5d1e711d6428e4d7ff210cbb7b5563d70843ce95274
Git SHA-1:                              5740078f3876277709b3259c9502ce01308b243a
Configuration directory:                /home/matthew/.mina-config
Peers:                                  0
User_commands sent:                     0
SNARK worker:                           None
SNARK work fee:                         1000000000
Sync status:                            Bootstrap
Block producers running:                0
Consensus time now:                     epoch=57369, slot=356
Consensus mechanism:                    proof_of_stake
Consensus configuration:
        Delta:                     0
        k:                         24
        Slots per epoch:           576
        Slot duration:             2s
        Epoch duration:            19m12s
        Chain start timestamp:     2019-01-30 20:00:00.000000Z
        Acceptable network delay:  2s

Addresses and ports:
        External IP:    89.242.88.249
        Bind IP:        0.0.0.0
        Libp2p PeerID:  12D3KooWSdazhA6kFTp9SPT8JutTx1AP33LyPDPoY2YkY4KHC8UN
        Libp2p port:    8302
        Client port:    8301


$ _build/default/src/app/cli/src/mina.exe advanced set-coinbase-receiver --public-key B62qrecVjpoZ4Re3a5arN6gXZ6orhmj1enUtA887XdG5mtZfdUbBUh4
Was sending coinbases to the block producer
Now sending coinbases to the public key B62qrecVjpoZ4Re3a5arN6gXZ6orhmj1enUtA887XdG5mtZfdUbBUh4
$ _build/default/src/app/cli/src/mina.exe client status
Mina daemon status
-----------------------------------

Max observed block height:              1
Max observed unvalidated block height:  0
Local uptime:                           25s
Chain id:                               971ab7c2a9370f90a3e4f5d1e711d6428e4d7ff210cbb7b5563d70843ce95274
Git SHA-1:                              5740078f3876277709b3259c9502ce01308b243a
Configuration directory:                /home/matthew/.mina-config
Peers:                                  0
User_commands sent:                     0
SNARK worker:                           None
SNARK work fee:                         1000000000
Sync status:                            Bootstrap
Block producers running:                0
coinbase receiver:                      B62qrecVjpoZ4Re3a5arN6gXZ6orhmj1enUtA887XdG5mtZfdUbBUh4
Consensus time now:                     epoch=57369, slot=364
Consensus mechanism:                    proof_of_stake
Consensus configuration:
        Delta:                     0
        k:                         24
        Slots per epoch:           576
        Slot duration:             2s
        Epoch duration:            19m12s
        Chain start timestamp:     2019-01-30 20:00:00.000000Z
        Acceptable network delay:  2s

Addresses and ports:
        External IP:    89.242.88.249
        Bind IP:        0.0.0.0
        Libp2p PeerID:  12D3KooWSdazhA6kFTp9SPT8JutTx1AP33LyPDPoY2YkY4KHC8UN
        Libp2p port:    8302
        Client port:    8301


$ _build/default/src/app/cli/src/mina.exe advanced set-coinbase-receiver --block-producer
Was sending coinbases to the public key B62qrecVjpoZ4Re3a5arN6gXZ6orhmj1enUtA887XdG5mtZfdUbBUh4
Now sending coinbases to the block producer
$ _build/default/src/app/cli/src/mina.exe client status
Mina daemon status
-----------------------------------

Max observed block height:              1
Max observed unvalidated block height:  0
Local uptime:                           38s
Chain id:                               971ab7c2a9370f90a3e4f5d1e711d6428e4d7ff210cbb7b5563d70843ce95274
Git SHA-1:                              5740078f3876277709b3259c9502ce01308b243a
Configuration directory:                /home/matthew/.mina-config
Peers:                                  0
User_commands sent:                     0
SNARK worker:                           None
SNARK work fee:                         1000000000
Sync status:                            Bootstrap
Block producers running:                0
Consensus time now:                     epoch=57369, slot=370
Consensus mechanism:                    proof_of_stake
Consensus configuration:
        Delta:                     0
        k:                         24
        Slots per epoch:           576
        Slot duration:             2s
        Epoch duration:            19m12s
        Chain start timestamp:     2019-01-30 20:00:00.000000Z
        Acceptable network delay:  2s

Addresses and ports:
        External IP:    89.242.88.249
        Bind IP:        0.0.0.0
        Libp2p PeerID:  12D3KooWSdazhA6kFTp9SPT8JutTx1AP33LyPDPoY2YkY4KHC8UN
        Libp2p port:    8302
        Client port:    8301
```

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

Closes #8125

### CODE REVIEW

File: src/lib/daemon_rpcs/types.ml [hunk 1]
```diff
@@ -250,6 +250,8 @@ module Status = struct
     let block_production_keys =
       list_string_entry "Block producers running" ~to_string:Fn.id
 
+    let coinbase_receiver = string_option_entry "coinbase receiver"
```
Comment (psteckler) [inline src/lib/daemon_rpcs/types.ml:253]: Should this be init-capped, as for the other status items?

**Comment by psteckler** [comment]
Maybe it would be better to always show the coinbase receiver in the status, rather than relying on users' implicit knowledge that it's the block producer when it's not mentioned. the status item could be "Coinbase receiver: block producer".

**Review by psteckler — APPROVED** [review]
Couple of nits, otherwise fine.

File: src/lib/daemon_rpcs/types.ml (hunk 1 above)
Comment (mrmr1993) [reply src/lib/daemon_rpcs/types.ml:253]: I'm not sure what this means

Comment (psteckler) [reply src/lib/daemon_rpcs/types.ml:253]: "Coinbase receiver"

**Comment by mrmr1993** [comment]
Updated. Output now looks like

```
$ _build/default/src/app/cli/src/mina.exe client status
Mina daemon status
-----------------------------------

Max observed block height:              1
Max observed unvalidated block height:  0
Local uptime:                           7s
Chain id:                               971ab7c2a9370f90a3e4f5d1e711d6428e4d7ff210cbb7b5563d70843ce95274
Git SHA-1:                              f1c3d015a3d23934a77c04b91a62239593a155a7
Configuration directory:                /home/matthew/.mina-config
Peers:                                  0
User_commands sent:                     0
SNARK worker:                           None
SNARK work fee:                         1000000000
Sync status:                            Bootstrap
Block producers running:                0
Coinbase_receiver:                      Block producer
Consensus time now:                     epoch=57374, slot=111
Consensus mechanism:                    proof_of_stake
Consensus configuration:
        Delta:                     0
        k:                         24
        Slots per epoch:           576
        Slot duration:             2s
        Epoch duration:            19m12s
        Chain start timestamp:     2019-01-30 20:00:00.000000Z
        Acceptable network delay:  2s

Addresses and ports:
        External IP:    89.242.88.249
        Bind IP:        0.0.0.0
        Libp2p PeerID:  12D3KooWCppenioydUHiPSx5ccfsDb8GMqkATrc7GqnUx12Zgeab
        Libp2p port:    8302
        Client port:    8301


$ _build/default/src/app/cli/src/mina.exe advanced set-coinbase-receiver --public-key B62qrecVjpoZ4Re3a5arN6gXZ6orhmj1enUtA887XdG5mtZfdUbBUh4
Was sending coinbases to the block producer
Now sending coinbases to the public key B62qrecVjpoZ4Re3a5arN6gXZ6orhmj1enUtA887XdG5mtZfdUbBUh4
$ _build/default/src/app/cli/src/mina.exe client status
Mina daemon status
-----------------------------------

Max observed block height:              1
Max observed unvalidated block height:  0
Local uptime:                           7s
Chain id:                               971ab7c2a9370f90a3e4f5d1e711d6428e4d7ff210cbb7b5563d70843ce95274
Git SHA-1:                              f1c3d015a3d23934a77c04b91a62239593a155a7
Configuration directory:                /home/matthew/.mina-config
Peers:                                  0
User_commands sent:                     0
SNARK worker:                           None
SNARK work fee:                         1000000000
Sync status:                            Bootstrap
Block producers running:                0
Coinbase_receiver:                      B62qrecVjpoZ4Re3a5arN6gXZ6orhmj1enUtA887XdG5mtZfdUbBUh4
Consensus time now:                     epoch=57374, slot=111
Consensus mechanism:                    proof_of_stake
Consensus configuration:
        Delta:                     0
        k:                         24
        Slots per epoch:           576
        Slot duration:             2s
        Epoch duration:            19m12s
        Chain start timestamp:     2019-01-30 20:00:00.000000Z
        Acceptable network delay:  2s

Addresses and ports:
        External IP:    89.242.88.249
        Bind IP:        0.0.0.0
        Libp2p PeerID:  12D3KooWCppenioydUHiPSx5ccfsDb8GMqkATrc7GqnUx12Zgeab
        Libp2p port:    8302
        Client port:    8301


$ _build/default/src/app/cli/src/mina.exe advanced set-coinbase-receiver --block-producer
Was sending coinbases to the public key B62qrecVjpoZ4Re3a5arN6gXZ6orhmj1enUtA887XdG5mtZfdUbBUh4
Now sending coinbases to the block producer
matthew@matthew-HP-ENVY-17-Notebook-PC:~/source_code/mina$ _build/default/src/app/cli/src/mina.exe client status
Mina daemon status
-----------------------------------

Max observed block height:              1
Max observed unvalidated block height:  0
Local uptime:                           7s
Chain id:                               971ab7c2a9370f90a3e4f5d1e711d6428e4d7ff210cbb7b5563d70843ce95274
Git SHA-1:                              f1c3d015a3d23934a77c04b91a62239593a155a7
Configuration directory:                /home/matthew/.mina-config
Peers:                                  0
User_commands sent:                     0
SNARK worker:                           None
SNARK work fee:                         1000000000
Sync status:                            Bootstrap
Block producers running:                0
Coinbase_receiver:                      Block producer
Consensus time now:                     epoch=57374, slot=111
Consensus mechanism:                    proof_of_stake
Consensus configuration:
        Delta:                     0
        k:                         24
        Slots per epoch:           576
        Slot duration:             2s
        Epoch duration:            19m12s
        Chain start timestamp:     2019-01-30 20:00:00.000000Z
        Acceptable network delay:  2s

Addresses and ports:
        External IP:    89.242.88.249
        Bind IP:        0.0.0.0
        Libp2p PeerID:  12D3KooWCppenioydUHiPSx5ccfsDb8GMqkATrc7GqnUx12Zgeab
        Libp2p port:    8302
        Client port:    8301
```

**Comment by psteckler** [comment]
Could the item be "Coinbase receiver" (no underscore), instead of `Coinbase_receiver", to match the other items?

File: src/lib/mina_lib/mina_lib.ml [hunk 2]
```diff
@@ -137,6 +137,11 @@ let block_production_pubkeys t : Public_key.Compressed.Set.t =
   let public_keys, _ = Agent.get t.block_production_keypairs in
   Public_key.Compressed.Set.map public_keys ~f:snd
 
+let coinbase_receiver t = !(t.coinbase_receiver)
+
+let replace_coinbase_receiver t coinbase_receiver =
+  t.coinbase_receiver := coinbase_receiver
```
Comment (deepthiskumar) [inline src/lib/mina_lib/mina_lib.ml:143]: Add an info log here when the change happens?

**Review by deepthiskumar — APPROVED** [review]
Awesome!

**Review by bkase — APPROVED** [review]
Agree with @deepthiskumar 's suggestion. Otherwise ✅

**Comment by bkase** [comment]
!approved-for-mainnet

**Pull request merged**